plugins/org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java

```diff
@@ -164,6 +169,42 @@ public void apply(final EObject element, IModificationContext context) {
 					}
 				});
 	}
+	@Fix(SPACES_IN_KEYWORD)
+	public void fixSpacesInKeyword(final Issue issue, IssueResolutionAcceptor acceptor) {
+		if (issue.getData().length > 0 && Boolean.valueOf(issue.getData()[0]))
+		acceptor.accept(issue, "Replace Keyword with multiple Keywords", "Splits a Keyword that contains spaces or tabs into multiple Keywords", NULL_QUICKFIX_IMAGE,
+				new ISemanticModification() {
```

Comment:
Don't you think a textual modification would be simpler, also for the case where you have to insert perentheses?